parsers/shared/src/main/scala/sigmastate/lang/ContractTemplateParser.scala

```diff
@@ -0,0 +1,12 @@
+package sigmastate.lang
+
+import fastparse._
+import fastparse.NoWhitespace._
+
+case class ParsedContractTemplate(docs: ContractDoc, signature: ContractSignature)
+
+object ContractTemplateParser {
+  def parse(source: String): Parsed[ParsedContractTemplate] = fastparse.parse(source, parse(_))
+
+  def parse[_: P]: P[ParsedContractTemplate] = P(ContractDocParser.parse ~ "\n" ~ ContractSignatureParser.parse ~ " ".rep.? ~ "=" ~ " ".rep.? ~ &("{")).map(s => ParsedContractTemplate(s._1, s._2))
+}
```

Comment:
I suggest to rename the main object to ContractParser. Will it be used for template or somewhere else it is separate question. And also I think we don't need many object, just put all the code in ContractParser class/object